platform/android/java/app/src/com/godot/game/GodotApp.java

```diff
@@ -56,8 +64,36 @@ public class GodotApp extends GodotActivity {
 		}
 	}
 
+	// Shouldn't conflict with BaseGodotEditor.RUN_GAME_INFO etc
+	private final static int WINDOW_ID = 668;
+	private final static String TAG = GodotApp.class.getSimpleName();
+
+	private ArrayList<String> commandLineParams = new ArrayList<String>();
+
+	@Override
+	public int onNewGodotInstanceRequested(String[] args) {
+		// Launch a new activity
+		Log.d(TAG, "Restarting with parameters " + String.join(",", args));
+		Godot godot = getGodot();
+		Intent newInstance = new Intent()
+									 .setComponent(new ComponentName(this, GodotApp.class))
+									 .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+									 .putExtra(getEXTRA_COMMAND_LINE_PARAMS(), args);
+		if (godot != null) {
+			godot.destroyAndKillProcess(
+					() -> ProcessPhoenix.triggerRebirth(this, newInstance));
+		} else {
+			ProcessPhoenix.triggerRebirth(this, newInstance);
+		}
+		return WINDOW_ID;
+	}
+
 	@Override
 	public void onCreate(Bundle savedInstanceState) {
+		Intent intent = getIntent();
+		String[] params = intent.getStringArrayExtra(getEXTRA_COMMAND_LINE_PARAMS());
+		Log.d(TAG, "Starting intent " + intent + " with parameters " + (params != null ? String.join(" ", params) : "null"));
+		updateCommandLineParams(params != null && params.length > 0 ? Arrays.asList(params) : Collections.emptyList());
```

Comment:
There is similar logic in the `onCreate` for `BaseGodotEditor` that can be moved to `GodotActivity` so we can avoid duplicating it in both classes.